resources/common/plugins/template/cvvc-oto-gen/cvvc-oto-gen.js

```diff
@@ -117,136 +112,322 @@ let aliasCountMap = new Map()
 
 let outputSampleCVMap = new Map()
 let outputSampleVCMap = new Map()
+let outputSampleOtherMap = new Map()
 
-function push(entry, asCV) {
+function push(entry, type) {
     if (!reorder) {
         output.push(entry)
         return
     }
 
     let sample = entry.sample
-    let map = asCV ? outputSampleCVMap : outputSampleVCMap
+    let map = type == "CV" ? outputSampleCVMap : type == "VC" ? outputSampleVCMap : outputSampleOtherMap
 
     let list = map.get(sample) || []
     list.push(entry)
     map.set(sample, list)
 }
 
-// Include CV, VCV, VV and special
-function pushCV(sample, index, alias, isSpecial, tag) {
-    // check alias count
-    let max = repeat
+function checkAliasCount(alias, isOther, isSingleC) {
+    let max = isSingleC ? repeatC : repeat
     let count = aliasCountMap.get(alias) || 0
-    if (!isSpecial && count >= max) {
-        return
+    if (!isOther && count >= max) {
+        return ""
     }
 
     let thisCount = count + 1
     let thisAlias = alias
     if (thisCount > 1) {
         thisAlias += repeatSuffix.replaceAll("{number}", thisCount)
     }
+    aliasCountMap.set(alias, count + 1)
+    return thisAlias
+}
+
+function pushHeadCV(sample, alias, nextHasConsonant) {
+    let thisAlias = checkAliasCount(alias, false, false)
 
-    let start = offset + index * beatLength - preuCV
-    let end = start - cutoffCV
-    let fixed = start + fixedCV
-    let preu = start + preuCV
-    let ovl = start + ovlCV
+    let start = offset - consLength - 10
+    let ovl = offset - consLength
+    let preu = offset
+    let fixed = preu + fixBuffer
+    let cutoff = 0 - (10 + consLength + beatLength - ovlVC - (nextHasConsonant ? consLength : 0))
+    
+    let end = start - cutoff
     let points = [fixed, preu, ovl]
-    // for oto labeler plus, adding start again in the points
```

Comment:
This is quite outdated and thank you for removing it 👍 
So I think we can just do `let points = [fixed, preu, ovl, start]`?